From fa18d91134739048ace5cbcbf5b6d24930937926 Mon Sep 17 00:00:00 2001
From: root <root@qiaoliu-icelake-vdi.sh.intel.com>
Date: Tue, 6 Dec 2022 03:15:11 -0500
Subject: [PATCH] Added auto mouse move to avoid stop connection

---
 rdpy/ui/qt4.py | 30 ++++++++++++++++++++++++++++--
 1 file changed, 28 insertions(+), 2 deletions(-)

diff --git a/rdpy/ui/qt4.py b/rdpy/ui/qt4.py
index 237f974..dee59c4 100644
--- a/rdpy/ui/qt4.py
+++ b/rdpy/ui/qt4.py
@@ -22,7 +22,9 @@ Qt specific code
 
 QRemoteDesktop is a widget use for render in rdpy
 """
-
+import random
+from PyQt4.QtCore import *
+from PyQt4.QtGui import QCursor
 from PyQt4 import QtGui, QtCore
 from rdpy.protocol.rfb.rfb import RFBClientObserver
 from rdpy.protocol.rdp.rdp import RDPClientObserver
@@ -89,6 +91,9 @@ class RFBClientQt(RFBClientObserver, QAdaptor):
         """
         RFBClientObserver.__init__(self, controller)
         self._widget = QRemoteDesktop(1024, 800, self)
+        self.timer = QTimer()
+        self.timer.timeout.connect(self.MouseMove)
+        self.timer.start(30000)
         
     def getWidget(self):
         """
@@ -179,6 +184,15 @@ class RFBClientQt(RFBClientObserver, QAdaptor):
         #do something maybe a message
         pass
 
+    def MouseMove(self):
+        winWidth = self._widget.geometry().width()
+        winHeight = self._widget.geometry().height()
+        xpos=random.randint(self._widget.geometry().x(),self._widget.geometry().x()+winWidth)
+        ypos=random.randint(self._widget.geometry().y(),self._widget.geometry().y()+winHeight)
+        cursor = QCursor()
+        cursor.setPos(xpos,ypos)
+
+
 def RDPBitmapToQtImage(width, height, bitsPerPixel, isCompress, data):
     """
     @summary: Bitmap transformation to Qt object
@@ -241,6 +255,10 @@ class RDPClientQt(RDPClientObserver, QAdaptor):
         self._widget = QRemoteDesktop(width, height, self)
         #set widget screen to RDP stack
         controller.setScreen(width, height)
+        self.timer = QTimer()
+        self.timer.timeout.connect(self.MouseMove)
+        self.timer.start(30000)
+
         
     def getWidget(self):
         """
@@ -329,6 +347,14 @@ class RDPClientQt(RDPClientObserver, QAdaptor):
         """
         #do something maybe a message
 
+    def MouseMove(self):
+        winWidth = self._widget.geometry().width()
+        winHeight = self._widget.geometry().height()
+        xpos=random.randint(self._widget.geometry().x(),self._widget.geometry().x()+winWidth)
+        ypos=random.randint(self._widget.geometry().y(),self._widget.geometry().y()+winHeight)
+        cursor = QCursor()
+        cursor.setPos(xpos,ypos)
+
         
 class QRemoteDesktop(QtGui.QWidget):
     """
@@ -428,4 +454,4 @@ class QRemoteDesktop(QtGui.QWidget):
         @summary: Call when widget is closed
         @param event: QCloseEvent
         """
-        self._adaptor.closeEvent(event)
\ No newline at end of file
+        self._adaptor.closeEvent(event)
-- 
1.8.3.1

