From 98a00c25303eb597a03be89109ffa4b3d1029941 Mon Sep 17 00:00:00 2001
From: root <root@qiaoliu-icelake-vdi.sh.intel.com>
Date: Tue, 7 Mar 2023 22:25:04 -0500
Subject: [PATCH 2/2] Add username and password support for capture rdp

---
 bin/rdpy-rdpscreenshot.py | 23 +++++++++++++++++------
 1 file changed, 17 insertions(+), 6 deletions(-)

diff --git a/bin/rdpy-rdpscreenshot.py b/bin/rdpy-rdpscreenshot.py
index b87008a..87abe89 100755
--- a/bin/rdpy-rdpscreenshot.py
+++ b/bin/rdpy-rdpscreenshot.py
@@ -45,7 +45,7 @@ class RDPScreenShotFactory(rdp.ClientFactory):
     __INSTANCE__ = 0
     __STATE__ = []
 
-    def __init__(self, reactor, app, width, height, path, timeout):
+    def __init__(self, reactor, app, width, height, path, username, password, timeout):
         """
         @param reactor: twisted reactor
         @param width: {integer} width of screen
@@ -58,6 +58,8 @@ class RDPScreenShotFactory(rdp.ClientFactory):
         self._app = app
         self._width = width
         self._height = height
+        self._username = username
+        self._passwod = password
         self._path = path
         self._timeout = timeout
         #NLA server can't be screenshooting
@@ -155,12 +157,13 @@ class RDPScreenShotFactory(rdp.ClientFactory):
 
             def checkUpdate(self):
                 self._controller.close();
-
+        controller.setUsername(self._username)
+        controller.setPassword(self._passwod)
         controller.setScreen(self._width, self._height);
         controller.setSecurityLevel(self._security)
         return ScreenShotObserver(controller, self._width, self._height, self._path, self._timeout, self._reactor)
 
-def main(width, height, path, timeout, hosts):
+def main(width, height, path, username, password, timeout, hosts):
     """
     @summary: main algorithm
     @param height: {integer} height of screenshot
@@ -184,7 +187,7 @@ def main(width, height, path, timeout, hosts):
         else:
             ip, port = host, "3389"
 
-        reactor.connectTCP(ip, int(port), RDPScreenShotFactory(reactor, app, width, height, path + "%s.jpg" % ip, timeout))
+        reactor.connectTCP(ip, int(port), RDPScreenShotFactory(reactor, app, width, height, path + "%s.jpg" % ip, username, password, timeout))
 
     reactor.runReturn()
     app.exec_()
@@ -195,6 +198,8 @@ def help():
     print "Usage: rdpy-rdpscreenshot [options] ip[:port]"
     print "\t-w: width of screen default value is 1024"
     print "\t-l: height of screen default value is 800"
+    print "\t-u: username of the rdp connection"
+    print "\t-p: password of the rdp connection"    
     print "\t-o: file path of screenshot default(/tmp/rdpy-rdpscreenshot.jpg)"
     print "\t-t: timeout of connection without any updating order (default is 2s)"
 
@@ -202,11 +207,13 @@ if __name__ == '__main__':
     # default script argument
     width = 1024
     height = 800
+    username = "administrator"
+    password = ""
     path = "/tmp/"
     timeout = 5.0
 
     try:
-        opts, args = getopt.getopt(sys.argv[1:], "hw:l:o:t:")
+        opts, args = getopt.getopt(sys.argv[1:], "hw:l:o:t:u:p:")
     except getopt.GetoptError:
         help()
     for opt, arg in opts:
@@ -221,5 +228,9 @@ if __name__ == '__main__':
             path = arg
         elif opt == "-t":
             timeout = float(arg)
+        elif opt == "-u":
+            username = arg
+        elif opt == "-p":
+            password = arg           
 
-    main(width, height, path, timeout, args)
+    main(width, height, path, username, password, timeout, args)
-- 
1.8.3.1

